core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -395,6 +401,14 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
 
         return HttpResponse.from(updateHandler.handle(ctx, req).thenApply(updateResult -> {
             if (updateResult != null) {
+                for (HealthCheckUpdateListener updateListener : updateListeners) {
```

Comment:
I think we can just add the listeners to the `serverHealth` in the constructor because `SettableHealthChecker` is a `AbstractListenable<HealthChecker>`.

Reply:
If add listeners to the `serverHealth`, update events are notified not only from `updateHandler` but also `ServerListener`. If then, I think that `HealthCheckUpdateListener` should be changed:
```java
/**
 * A listener interface for receiving {@link HealthCheckUpdateHandler} events.
 */
@FunctionalInterface
public interface HealthCheckUpdateListener {
```

I'm not sure which behavior is better, accepting update events including server startup or just from `updateHandler`. Could you give me some advice considering this?

Reply:
I think the listener should listen to all events to the `HealthChecker`.

So, you want to control updating the healthy state of the server not by the Armeria but by the `HealthCheckUpdateHandler`. So you are using another `HealthChecker` whose healthy is set to true when it's updating by `HealthCheckUpdateHandler`, right?

Then, how about adding a method that disables `serverHealth.setHealthy(true);` from the `ServerListner`? If so, we don't even need to add an additional `HealthChecker`.
```
public HealthCheckServiceBuilder disableServerListenerUpdate() {
```

Reply:
Nice!